p2p/peering.go

```diff
@@ -181,7 +181,7 @@ func newNetAddress(addr net.Addr, services wire.ServiceFlag) (*wire.NetAddress,
 		return nil, fmt.Errorf("newNetAddress: unsupported address "+
 			"type %T", addr)
 	}
-	if ip.IsLoopback() || ip.IsPrivate() {
+	if !ip.IsGlobalUnicast() {
```

Comment:
I think you'd want to not bother sending private addresses back either which this doesn't do.

```Go
package main

import (
	"fmt"
	"net"
)

func main() {
	ipv4Private1 := net.ParseIP("10.0.0.1")
	ipv4Private2 := net.ParseIP("172.16.0.1")
	ipv4Private3 := net.ParseIP("192.168.0.1")
	ipv6Private1 := net.ParseIP("fc00::1")

	p := func(ip net.IP) {
		fmt.Printf("%v (IsGlobalUnicast: %v, IsPrivate: %v)\n", ip, ip.IsGlobalUnicast(), ip.IsPrivate())
	}

	p(ipv4Private1)
	p(ipv4Private2)
	p(ipv4Private3)
	p(ipv6Private1)
}
```

Output:
```
10.0.0.1 (IsGlobalUnicast: true, IsPrivate: true)
172.16.0.1 (IsGlobalUnicast: true, IsPrivate: true)
192.168.0.1 (IsGlobalUnicast: true, IsPrivate: true)
fc00::1 (IsGlobalUnicast: true, IsPrivate: true)
```

Reply:
Meh, you already updated it in the next commit.